File: Backend/BaseMgtService/Senior.AgileAI.BaseMgt.Application/DTOs/Meetings/ModifyRecurringMeetingDto.cs
using Senior.AgileAI.BaseMgt.Domain.Enums;

namespace Senior.AgileAI.BaseMgt.Application.DTOs.Meetings;

public class ModifyRecurringMeetingDto
{
    public required bool ApplyToSeries { get; init; }
    public string? Title { get; init; }
    public string? Goal { get; init; }
    public MeetingLanguage? Language { get; init; }
    public DateTime? StartTime { get; init; }
    public DateTime? EndTime { get; init; }
    public string? TimeZone { get; init; }
    public string? Location { get; init; }
    public DateTime? ReminderTime { get; init; }
    public MeetingStatus? Status { get; init; }
    public List<Guid>? AddMembers { get; init; }
    public List<Guid>? RemoveMembers { get; init; }
} 

File: agilemeets_mobile/lib/data/models/modify_recurring_meeting_dto.dart
import 'package:agilemeets/data/enums/meeting_language.dart';
import 'package:agilemeets/data/enums/meeting_status.dart';

class ModifyRecurringMeetingDTO {
  final bool applyToSeries;
  final String? title;
  final String? goal;
  final MeetingLanguage? language;
  final DateTime? startTime;
  final DateTime? endTime;
  final String? timeZone;
  final String? location;
  final DateTime? reminderTime;
  final MeetingStatus? status;
  final List<String>? addMembers;
  final List<String>? removeMembers;

  const ModifyRecurringMeetingDTO({
    required this.applyToSeries,
    this.title,
    this.goal,
    this.language,
    this.startTime,
    this.endTime,
    this.timeZone,
    this.location,
    this.reminderTime,
    this.status,
    this.addMembers,
    this.removeMembers,
  });

  Map<String, dynamic> toJson() => {
    'applyToSeries': applyToSeries,
    if (title != null) 'title': title,
    if (goal != null) 'goal': goal,
    if (language != null) 'language': language?.value,
    if (startTime != null) 'startTime': startTime?.toIso8601String(),
    if (endTime != null) 'endTime': endTime?.toIso8601String(),
    if (timeZone != null) 'timeZone': timeZone,
    if (location != null) 'location': location,
    if (reminderTime != null) 'reminderTime': reminderTime?.toIso8601String(),
    if (status != null) 'status': status?.value,
    if (addMembers != null) 'addMembers': addMembers,
    if (removeMembers != null) 'removeMembers': removeMembers,
  };
} 